src/xpcc/architecture/platform/driver/core/cortex/driver.xml

```diff
@@ -38,11 +39,22 @@
 		<!-- stm32f7 -->
 		<template device-family="f7" out="linkerscript.ld">stm32/stm32_idtcm.ld.in</template>
 
-		<template>startup.c.in</template>
-		<template>delay.cpp.in</template>
-		<template>syscalls.cpp.in</template>
+		<!-- dealing with C++ -->
 		<static>cxxabi.cpp</static>
+		<!-- dealing with newlib -->
+		<static>syscalls.c</static>
+		<!-- startup script -->
+		<template>startup.c.in</template>
 		<static>default_handler.sx</static>
+		<!-- everything to do with dynamic memory -->
+		<static>heap_table.c</static>
+		<template>heap_newlib.c.in</template>
```

Comment:
Is it possible to filter for parameter settings?
Something like:

<template parameter-allocator="newlib">heap_newlib.c.in</template>
Maybe this should be a python file. XML seems the wrong format. cc @ekiwi